tests/projects/test_project.py

```diff
@@ -470,6 +474,21 @@ def test_set_func_with_tag():
     assert func.metadata.tag is None
 
 
+def test_set_function_with_relative_path(context):
```

Comment:
was this failing *before* your change?

Reply:
This was never meant to fail, I added it because I noticed we didn't cover this case for set function, and I was already adding coverage for import_artifact.
It was working before my change because the logic was called from '_init function from dict'.